### PR TITLE
feat: EnforceWithMatcher & EnforceExWithMatcher

### PR DESCRIPTION
**Pull request by Shivansh-yadav13**
- added enforceWithMatcher & enforceExWithMatcher

### CODE REVIEW

File: src/coreEnforcer.ts [hunk 1]
```diff
@@ -400,9 +401,14 @@ export class CoreEnforcer {
       functions[key] = asyncCompile ? generateGFunction(rm) : generateSyncedGFunction(rm);
     });
 
-    const expString = this.model.model.get('m')?.get('m')?.value;
-    if (!expString) {
-      throw new Error('Unable to find matchers in model');
+    let expString;
+    if (!matcher) {
+      expString = this.model.model.get('m')?.get('m')?.value;
+      if (!expString) {
```
Comment (Sefriol) [inline src/coreEnforcer.ts:407]: Move this check after the else condition when the matcher exists.

**Comment by hsluoyz** [comment]
@Sefriol

File: test/enforcer.test.ts [hunk 2]
```diff
@@ -697,3 +697,10 @@ test('TestEnforceExWithPriorityModel', async () => {
   testEnforceEx(e, 'bob', 'data2', 'read', [true, ['data2_allow_group', 'data2', 'read', 'allow']]);
   testEnforceEx(e, 'alice', 'data2', 'read', [false, []]);
 });
+
```
Comment (Sefriol) [inline test/enforcer.test.ts:700]: This should test all the normal test cases for this model + policy and verify all the other ramifications with an alternative matcher. Single test enforce can be flagged true or false out of the randomness of a faulty pull request.

Comment (nodece) [reply test/enforcer.test.ts:700]: Could you review this PR once?

**Comment by hsluoyz** [comment]
Replaced by: https://github.com/casbin/node-casbin/pull/453

**Pull request closed without merging**